flow/flat_buffers.h

```diff
@@ -129,6 +130,33 @@ struct vector_like_traits<std::vector<T, Alloc>> : std::true_type {
 	}
 };
 
+template <class T, class Alloc>
+struct vector_like_traits<std::deque<T, Alloc>> : std::true_type {
+	using Deq = std::deque<T, Alloc>;
+	using value_type = typename Deq::value_type;
+	using iterator = typename Deq::const_iterator;
+	using insert_iterator = std::back_insert_iterator<Deq>;
+
+	template <class Context>
+	static size_t num_entries(const Deq& v, Context&) {
+		return v.size();
+	}
+	template <class Context>
+	static void reserve(Deq& v, size_t size, Context&) {
+		v.resize(size);
+		v.clear();
```

Comment:
Why is the `clear()` needed here?

Reply:
In order to reset the size to 0 after using `resize(size)`.
I assumed here 'reserve` should have the same effect as `vector_like_traits<std::vector<T, Alloc>>`when I implemented it, but `deque` doesn't have `reserve` method, so I have to use `resize(size)` plus `clear`.
It might be better to change both `vector_like_traits<std::vector<T, Alloc>>` and `vector_like_traits<std::deque<T, Alloc>>` to match the cpp `reserve` semantic...

Reply:
Ok I see the discussion in #6681